cypress/support/assertions/expect_dual_list.js

```diff
@@ -0,0 +1,227 @@
+ * This flow assumes items start in the right list and tests moving them to the left and back
+ *
+ * @param {string[]} initialSelectedItems - Array of selected items texts to test with
+ * @returns {void}
+ */
+function testDualListStartingFromSelectedItems(initialSelectedItems) {
+  // Test moving a single item from right to left and back
+  selectAndMoveItem('right', 'left', initialSelectedItems[0]);
+  selectAndMoveItem('left', 'right', initialSelectedItems[0]);
+  // Test moving all items from right to left
+  moveAllItems('right', 'left', initialSelectedItems);
+  // Test search functionality on left list
+  testSearchFunctionality('left', initialSelectedItems[0]);
+  // Test moving all items from left to right
+  moveAllItems('left', 'right', initialSelectedItems);
+  // Test search functionality on right list
+  testSearchFunctionality('right', initialSelectedItems[0]);
+}
+
+/**
+ * Cypress command to test a dual-list component
+ *
+ * This command tests all aspects of a dual-list component including:
+ * - Initial state validation
+ * - Selecting and moving individual items between lists
+ * - Moving all items between lists
+ * - Search functionality in both lists
+ *
+ * The command automatically detects whether to test a left-to-right flow or right-to-left flow
+ * based on which list has items initially.
+ *
+ * @example
+ * // Test a dual-list with items initially in the available items
+ * cy.expect_dual_list({
+ *   availableItemsHeaderText: 'Available Items',
+ *   selectedItemsHeaderText: 'Selected Items',
+ *   availableItems: ['Item 1', 'Item 2', 'Item 3']
+ * });
+ *
+ * @example
+ * // Test a dual-list with items initially in the selected items
+ * cy.expect_dual_list({
+ *   availableItemsHeaderText: 'Unassigned Roles',
+ *   selectedItemsHeaderText: 'Assigned Roles',
+ *   selectedItems: ['Role 1', 'Role 2', 'Role 3']
+ * });
+ */
+Cypress.Commands.add(
+  'expect_dual_list',
+  ({
+    availableItemsHeaderText,
+    selectedItemsHeaderText,
+    availableItems = [],
+    selectedItems = [],
+  }) => {
+    if (availableItems.length < 1 && selectedItems.length < 1) {
+      cy.logAndThrowError(
+        'availableItems or selectedItems should have at least 1 item'
+      );
+    }
+
+    // Verify headings
+    if (availableItemsHeaderText) {
+      cy.contains('h6', availableItemsHeaderText).should('be.visible');
+    }
+    if (selectedItemsHeaderText) {
+      cy.contains('h6', selectedItemsHeaderText).should('be.visible');
+    }
+
+    // Determine which flow to test based on which list has items initially
+    if (availableItems.length) {
+      // Verify initial items in the left list
+      verifyListItems('first', availableItems);
+      // Left-to-right flow (when left list has items)
+      testDualListStartingFromAvailableItems(availableItems);
+    } else if (selectedItems.length) {
+      // Verify initial items in the right list
+      verifyListItems('last', selectedItems);
```

Comment:
I would prefer validate over verify... Assert would be good too... I think validate/assert implies not only verification but assertions that it will fail if not met.  Is this testing the list items contains the selectedItems or IS the selectedItems?  In other words, can there be other items in addition to the selected items...

We can probably merge as is but would like your feedback so we have direction on what to do in a possible followup PR.